### PR TITLE
Cl league stats

### PR DESCRIPTION
**Pull request by cheeleertr**
I wrote a lot of tests and created all the methods for the LeagueStats module. Wrote a lot of helper methods. Not sure if all the helper methods make the code easier or harder to read but it made writing the code easier. Some of the tests didn't like having numbers in the hashes so I made them "to be_a Hash" instead. I commented out the actual hash results under the test.

### CODE REVIEW

File: lib/league_stats.rb [hunk 1]
```diff
@@ -1,4 +1,127 @@
 
 
 module LeagueStats
+  def count_of_teams
```
Comment (jdmchugh111) [inline lib/league_stats.rb:4]: great work on these

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -7,6 +7,7 @@
 require './lib/season_stats'
 
 class StatTracker
+    attr_reader :games, :teams, :game_teams
```
Comment (jdmchugh111) [inline lib/stat_tracker.rb:10]: not sure if we need these attr_readers: unless you somehow are incorporating them in testing

File: spec/game_team_spec.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+require 'spec_helper.rb'
+
+RSpec.describe Game do
```
Comment (jdmchugh111) [inline spec/game_team_spec.rb:3]: maybe want to change this from Game to GameTeam

File: spec/stat_tracker_spec.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
+require 'spec_helper.rb'
+
+RSpec.describe Game do
```
Comment (jdmchugh111) [inline spec/stat_tracker_spec.rb:3]: same thing here might want to say describe StatTracker instead of Game

File: spec/team_spec.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+require 'spec_helper.rb'
+
+RSpec.describe Team do
+  it "can initialize" do
```
Comment (jdmchugh111) [inline spec/team_spec.rb:4]: thank you for going above and beyond to do the spec tests for game, team, game_teams, and stat_tracker. I knew someone was going to have to do that eventually!

File: lib/stat_tracker.rb (hunk 2 above)
Comment (cheeleertr) [reply lib/stat_tracker.rb:10]: I needed them for writing tests and some of the methods but didn't use the :game_teams attr_reader.

Comment (JCL461437) [reply lib/stat_tracker.rb:10]: Smart! That was a clever way to address the issue.

File: spec/team_spec.rb (hunk 5 above)
Comment (JCL461437) [reply spec/team_spec.rb:4]: Yeah thanks Chee. Your a legend man.

**Pull request closed without merging**